PopupBridge/src/main/java/com/braintreepayments/api/PopupBridgeClient.kt

```diff
@@ -46,92 +51,90 @@ class PopupBridgeClient @SuppressLint("SetJavaScriptEnabled") @VisibleForTesting
 
         webView.settings.javaScriptEnabled = true
         webView.addJavascriptInterface(this, POPUP_BRIDGE_NAME)
-
-        popupBridgeLifecycleObserver.onBrowserSwitchResult = { result -> onBrowserSwitchResult(result) }
-        activity.lifecycle.addObserver(popupBridgeLifecycleObserver)
     }
 
-    private fun runJavaScriptInWebView(script: String) {
-        val webView = webViewRef.get() ?: return
-        webView.post(Runnable {
-            val webViewInternal = webViewRef.get() ?: return@Runnable
-            webViewInternal.evaluateJavascript(script, null)
-        })
+    fun handleReturnToApp(intent: Intent) {
+        val pendingRequest = pendingRequestRepository.getPendingRequest()
+        if (pendingRequest == null) return
+
+        pendingRequestRepository.clearPendingRequest()
+
+        when (val browserSwitchFinalResult = browserSwitchClient.completeRequest(intent, pendingRequest)) {
+            is BrowserSwitchFinalResult.Success -> runSuccessJavaScript(browserSwitchFinalResult.returnUrl)
+            is BrowserSwitchFinalResult.Failure -> runCanceledJavaScript()
```

Comment:
Should we run the "cancel" javascript when `completeRequest()` returns a Failure? It doesn't look like we previously had a failure case.

Reply:
🤔 that's a good question. We don't do anything with the [error returned from ASWeb on iOS currently](https://github.com/braintree/popup-bridge-ios/blob/0aa9ce097a79ba5099b19c65bdd8ca5a6fd64b82/Sources/PopupBridge/POPPopupBridge.swift#L147). Not sure if we want to just log the error to the console or something similar? Wondering if @richherrera has thoughts.

Reply:
If we have a failure scenario for both platforms maybe we should add `runFailureJavaScript()` to both?

Reply:
It might be worth adding error handling for the failure case so that the merchant can notify the user. I don't think we have any javascript code for this scenario. Would we be able to reach out to someone on the web side to help us with this?

Reply:
Since this will likely be a longer task (if possible) let's just mark a TODO to follow up, no need to block this PR on it

Reply:
I created an additional story to handle this failure case.